app/app/src/main/kotlin/com/hedvig/android/app/logginginterceptor/HedvigLoggingInterceptor.kt

```diff
@@ -0,0 +1,302 @@
+    newHeadersToRedact += headersToRedact
+    newHeadersToRedact += name
+    headersToRedact = newHeadersToRedact
+  }
+
+  /**
+   * Sets the level and returns this.
+   *
+   * This was deprecated in OkHttp 4.0 in favor of the [level] val. In OkHttp 4.3 it is
+   * un-deprecated because Java callers can't chain when assigning Kotlin vals. (The getter remains
+   * deprecated).
+   */
+  fun setLevel(level: Level) = apply {
+    this.level = level
+  }
+
+  @JvmName("-deprecated_level")
+  @Deprecated(
+    message = "moved to var",
+    replaceWith = ReplaceWith(expression = "level"),
+    level = DeprecationLevel.ERROR
+  )
+  fun getLevel(): Level = level
+
+  @Throws(IOException::class)
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val level = this.level
+
+    val request = chain.request()
+    if (level == Level.NONE) {
+      return chain.proceed(request)
+    }
+
+    val logBody = level == Level.BODY
+    val logHeaders = logBody || level == Level.HEADERS
+
+    val requestBody = request.body
+
+    val connection = chain.connection()
+    var requestStartMessage =
+      ("--> ${request.method} ${request.url}${if (connection != null) " " + connection.protocol() else ""}")
+    if (!logHeaders && requestBody != null) {
+      requestStartMessage += " (${requestBody.contentLength()}-byte body)"
+    }
+    logger.log(requestStartMessage)
+
+    if (logHeaders) {
+      val headers = request.headers
+
+      if (requestBody != null) {
+        // Request body headers are only present when installed as a network interceptor. When not
+        // already present, force them to be included (if available) so their values are known.
+        requestBody.contentType()?.let {
+          if (headers["Content-Type"] == null) {
+            logger.log("Content-Type: $it")
+          }
+        }
+        if (requestBody.contentLength() != -1L) {
+          if (headers["Content-Length"] == null) {
+            logger.log("Content-Length: ${requestBody.contentLength()}")
+          }
+        }
+      }
+
+      for (i in 0 until headers.size) {
+        logHeader(headers, i)
+      }
+
+      if (!logBody || requestBody == null) {
+        logger.log("--> END ${request.method}")
+      } else if (bodyHasUnknownEncoding(request.headers)) {
+        logger.log("--> END ${request.method} (encoded body omitted)")
+      } else if (requestBody.isDuplex()) {
+        logger.log("--> END ${request.method} (duplex request body omitted)")
+      } else if (requestBody.isOneShot()) {
+        logger.log("--> END ${request.method} (one-shot body omitted)")
+      } else {
+        val buffer = Buffer()
+        requestBody.writeTo(buffer)
+
+        val contentType = requestBody.contentType()
+        val charset: Charset = contentType?.charset(UTF_8) ?: UTF_8
+
+        logger.log("")
+        if (buffer.size > MAX_LOG_SIZE) {
+          logger.log("--> END ${requestBody.contentLength()}-byte body")
```

Comment:
So that's interesting, it is able to figure out what the size of the buffer is at this point, but it fails to do the `readString` on it since it is only at that point that it tries to put everything in memory. I don't quite understand how this works internally but if it can know about its size then we are all good here. I doubt we'd ever want to log these huge contents in any case, so simply omitting them entirely sounds like a good workaround for our needs!

Great job looking into this!